components/bpmn/org.wso2.carbon.bpmn.analytics.publisher/src/main/java/org/wso2/carbon/bpmn/analytics/publisher/BPMNDataPublisher.java

```diff
@@ -236,4 +340,182 @@ private String getProcessStreamId() {
+                String varName = configedProcessVariables[i][0];
+                String varType = configedProcessVariables[i][1];
+
+                Object varValue = variableInstances.get(varName).getValue();
+
+                switch (varType) {
+                case "int":
+                    if (varValue == null) {
+                        payload[i] = -1;
+                    } else {
+                        payload[i] = Integer.parseInt((String) varValue);
+                    }
+                    break;
+                case "float":
+                    if (varValue == null) {
+                        payload[i] = -1;
+                    } else {
+                        payload[i] = Float.parseFloat((String) varValue);
+                    }
+                    break;
+                case "long":
+                    if (varValue == null) {
+                        payload[i] = -1;
+                    } else {
+                        payload[i] = Long.parseLong((String) varValue);
+                    }
+                    break;
+                case "double":
+                    if (varValue == null) {
+                        payload[i] = -1;
+                    } else {
+                        payload[i] = Double.parseDouble((String) varValue);
+                    }
+                    break;
+                case "string":
+                    if (varValue == null) {
+                        payload[i] = "";
+                    } else {
+                        payload[i] = varValue;
+                    }
+                    break;
+                case "bool":
+                    if (varValue == null) {
+                        payload[i] = false;
+                    } else {
+                        payload[i] = Boolean.parseBoolean((String) varValue);
+                    }
+                    break;
+                default:
+                    log.warn("Configured process variable type is not a WSO2 DAS applicable type for the process:"
+                            + processDefinitionId);
+                    if (varValue == null) {
+                        payload[i] = "";
+                    } else {
+                        payload[i] = varValue;
+                    }
+                    break;
+                }
+            }
+
+            //set process instance id as the last payload variable value
+            payload[configedProcessVariables.length - 1] = processInstanceId;
+
+            boolean dataPublishingSuccess = dataPublisher.tryPublish(eventStreamId, getMeta(), null, payload);
+            if (dataPublishingSuccess) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Published BPMN process instance KPI event...  Process Instance Id :" + processInstanceId
+                            + ", Process Definition Id:" + processDefinitionId + ", Published Event's Payload Data :"
+                            + payload.toString());
```

Comment:
IMO it is not good because if the payload contains sensitive data then it should not logged.